### PR TITLE
[Core] Fix override memory and object_store_memory in decorator

### LINKED ISSUE #10562: [Core] `ray.options(memory=)` not overriding default.
### What is the problem?

*Ray version and other system information (Python version, TensorFlow version, OS):*

### Reproduction (REQUIRED)
Please provide a script that can be run to reproduce the issue. The script should have **no external library dependencies** (i.e., use fake or mock data / environments):

```
>>> @ray.remote(memory=2**30)
... def foo():
...  print(f"cluster memory: {ray.cluster_resources()['memory']}")
...  print(f"available memory: {ray.available_resources()['memory']}")
... 
>>> ray.get(foo.remote())
(pid=41181) cluster memory: 103.0
(pid=41181) available memory: 82.0

>>> ray.get(foo.options(memory=2**31).remote())
(pid=41181) cluster memory: 103.0
(pid=41181) available memory: 82.0
```

If we cannot run your script, we cannot fix your issue.

- [ ] I have verified my script runs in a clean environment and reproduces the issue.
- [ ] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/latest/installation.html).

### PR DESCRIPTION
**Pull request by wuisawesome**
## Related issue number

Closes #10562 

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

**Review by ericl — APPROVED** [review]

**Comment by rkooo567** [comment]
`test_memory_scheduling` seems to fail on both Mac and Linux. Can you make sure they are not related to this PR?

File: rllib/evaluation/worker_set.py [hunk 1]
```diff
@@ -98,9 +98,10 @@ def add_workers(self, num_workers: int) -> None:
         remote_args = {
             "num_cpus": self._remote_config["num_cpus_per_worker"],
             "num_gpus": self._remote_config["num_gpus_per_worker"],
-            "memory": self._remote_config["memory_per_worker"],
-            "object_store_memory": self._remote_config[
-                "object_store_memory_per_worker"],
+            # memory=0 is an error, but memory=None means no limits.
```
Comment (wuisawesome) [inline rllib/evaluation/worker_set.py:101]: @sven1977 can you double check this is ok?

Comment (sven1977) [reply rllib/evaluation/worker_set.py:101]: Makes sense. Thanks for fixing this.

**Review by sven1977 — APPROVED** [review]
Thanks for the fix.

**Pull request merged**